Code/Framework/AzCore/AzCore/Debug/TraceReflection.cpp

```diff
@@ -48,63 +94,197 @@ namespace AZ
             bool OnPrintf(const char* window, const char* message) override;
             bool OnOutput(const char* window, const char* message) override;
 
+            // AZ::TickBus::Handler overrides ...
+            void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+            int GetTickOrder() override;
+
         private:
-            template<class R, class... Args>
-            R CallResultReturn(const R& defaultReturnValue, int index, Args&&... args) const
+            void QueueMessageCall(AZStd::function<void()> messageCall);
+            void FlushMessageCalls();
+
+            AZStd::list<AZStd::function<void()>> m_messageCalls;
+            AZStd::mutex m_lock;
+
+            const bool m_defaultReturnValue = false;
```

Comment:
is this leftover from debugging? can it be removed?

Reply:
It's not left over code, that's the list where the messages are queued, its mutex to protect the list insertion/flush and a const variable to keep the magic bool false that all the functions return.

Reply:
i was referring to the ```const bool m_defaultReturnValue = false;``` (i only had the one line selected, but i guess github expanded it....
not sure the named var adds much more value then 'return false'.

Reply:
I can remove it. It's not big deal.